conformance/README.md

```diff
@@ -608,33 +610,63 @@ expectation, ending that branch of the test tree.
-                |  string
-                |  "("  "null"    model-type    ")"
-                |  "("  "string"  codepoint*    ")"
-                |  "("  "symbol"  model-symtok  ")"
-                |  "("  "list"    model-value*  ")"
-                |  "("  "sexp"    model-value*  ")"
-                |  "("  "struct"  model-field*  ")"
-                |  "("  "blob"    bytes*        ")"
-                |  "("  "clob"    bytes*        ")"
-
-// TODO Other types per denotational semantics
-
-codepoint     ::=  int                               // in the range 0..0x10FFFF
-
-model-symtok  ::=  string
-                |  int
-                |  "("  "text"  codepoint*  ")"
-                |  "("  "absent"  string  int  ")"       // symtab name + offset
-
-model-field   ::=  "("  model-symtok  model-value  ")"
-
-annotated     ::=  "("  "annot"  model-content  string*  ")"
+model-value     ::=  model-content  |  annotated
+
+model-content   ::=  null.null
+                  |  bool
+                  |  int
+                  |  string
+                  |  "("  "null"      model-type       ")"
+                  |  "("  "bool"      bool             ")"
+                  |  "("  "int"       int              ")"
+                  |  "("  "float"     string           ")" // See https://amazon-ion.github.io/ion-docs/docs/float.html
+                  |  "("  "decimal"   model-decimal    ")"
+                  |  "("  "timestamp" model-timestamp  ")"
+                  |  "("  "string"    codepoint*       ")"
+                  |  "("  "symbol"    model-symtok     ")"
+                  |  "("  "list"      model-value*     ")"
+                  |  "("  "sexp"      model-value*     ")"
+                  |  "("  "struct"    model-field*     ")"
+                  |  "("  "blob"      bytes*           ")"
+                  |  "("  "clob"      bytes*           ")"
+
+codepoint       ::=  int                                   // in the range 0..0x10FFFF
+
+model-symtok    ::=  string
+                  |  int
+                  |  "("  "text"  codepoint*  ")"
+                  |  "("  "absent"  string  int  ")"       // symtab name + offset
+
+model-field     ::=  "("  model-symtok  model-value  ")"
+
+model-decimal   ::= int int                                // coefficient + exponent
+                  | "(" "negative_0" ")" int
+
+// All timestamp subfields are interpreted as UTC time.
+// I.e. the following timestamps are not equivalent, but they represent the same point in time.
+// (timestamp (precision second) 1 2 3 (offset -1440) 4 5 6)
+// (timestamp (precision second) 1 2 3 (offset +1440) 4 5 6)
+
+model-timestamp ::= prec-year     int
+                  | prec-month    int int
+                  | prec-day      int int int
+                  | prec-minute   int int int offset int int
+                  | prec-second   int int int offset int int int
+                  | prec-fraction int int int offset int int int model-decimal
```

Comment:
I think it would be sufficent to replace (eg) `prec-year` clause with a symbol here.  Eg `(timestamp year <int>)` and `(timestamp minute 2024 6 4 (offset -240) 5 22)`

That's enough to make these grammar clauses distinct and context-free.

Reply:
Note if we go this way, we can use your original proposal for (decimal negative_zero 1).   Now that we have more use cases for that pattern, I can get behind it.